src/widgets/chart/lv_chart.c

```diff
@@ -344,34 +344,42 @@ lv_chart_series_t * lv_chart_add_series(lv_obj_t * obj, lv_color_t color, lv_cha
 
     LV_ASSERT_OBJ(obj, MY_CLASS);
 
-    lv_chart_t * chart    = (lv_chart_t *)obj;
+    lv_chart_t * chart = (lv_chart_t *)obj;
+
+    /* Allocate space for a new series and add it to the chart series linked list */
     lv_chart_series_t * ser = _lv_ll_ins_head(&chart->series_ll);
     LV_ASSERT_MALLOC(ser);
     if(ser == NULL) return NULL;
 
-    lv_coord_t def = LV_CHART_POINT_NONE;
-
-    ser->color  = color;
+    /* Allocate memory for point_cnt points, handle failure below */
     ser->y_points = lv_malloc(sizeof(lv_coord_t) * chart->point_cnt);
     LV_ASSERT_MALLOC(ser->y_points);
 
     if(chart->type == LV_CHART_TYPE_SCATTER) {
         ser->x_points = lv_malloc(sizeof(lv_coord_t) * chart->point_cnt);
         LV_ASSERT_MALLOC(ser->x_points);
+        /* NOTE: Should we return if ser->x_points is NULL? Is this an out of memory error? */
```

Comment:
Yes,  I think so. There is no point to continue in that case. It'd be great to free the other `malloc`s too.

Reply:
Is d769df2d2 ok?

Reply:
Yes, looks great!